### PR TITLE
Remove dependency on tensorflow-addons

### PR DESCRIPTION
**Pull request by d61h6k4**
**Proposed changes**:
`tensorflow-addons` is deprecated. Read [here](https://www.tensorflow.org/addons) for more.

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/utils/tensorflow/crf.py
Comment (m-vdb) [inline rasa/utils/tensorflow/crf.py:1]: should we mention that the code is ported from tensorflow_addons? can we also port some tests?

**Review by m-vdb — COMMENTED** [review]
I think we need to discuss if we shouldn't target `3.6.x`. I'll raise this to Rosana

File: rasa/utils/tensorflow/crf.py
Comment (d61h6k4) [reply rasa/utils/tensorflow/crf.py:1]: There is a comment line that mentions the original source of the code in a similar way as [here](https://github.com/RasaHQ/rasa/blob/main/rasa/utils/tensorflow/crf.py#L9)
Let me know if you want to do it in another way.
Will port tests too

Comment (m-vdb) [reply rasa/utils/tensorflow/crf.py:1]: ok good -- I missed that, thanks!

File: rasa/utils/tensorflow/metrics.py [hunk 1]
```diff
@@ -0,0 +1,282 @@
+import tensorflow as tf
+from tensorflow.keras import backend as K
+from tensorflow.types.experimental import TensorLike
+from typing import Any, Dict, Optional
+
+
+# original code taken from
+# https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/f_scores.py
+# (modified to our neeeds)
```
Comment (twerkmeister) [inline rasa/utils/tensorflow/metrics.py:9]: Can we be more specific with the reference that we have taken the code parts from, e.g. using a tag or commit hash? If we ever need to check what we changed ourselves, that would be useful. Same for all the copies.

**Review by twerkmeister — APPROVED** [review]
Looks good, with one suggestion

**Pull request merged**